src/types/python/excelreader.rs

```diff
@@ -190,6 +207,53 @@ impl ExcelReader {
             }
         }
     }
+
+    #[allow(clippy::too_many_arguments)]
+    fn build_table(
+        &mut self,
+        name: String,
+        header_row: Option<usize>,
+        column_names: Option<Vec<String>>,
+        skip_rows: usize,
+        n_rows: Option<usize>,
+        schema_sample_rows: Option<usize>,
+        dtype_coercion: DTypeCoercion,
+        use_columns: Option<&Bound<'_, PyAny>>,
+        dtypes: Option<DTypeMap>,
+        eager: bool,
+        py: Python<'_>,
+    ) -> PyResult<PyObject> {
+        let selected_columns = Self::build_selected_columns(use_columns).into_pyresult()?;
+
+        let table = self.sheets.get_table(&name).into_pyresult()?;
+        let column_names: Vec<String> = {
+            match column_names {
+                None => Vec::from(table.columns()),
+                Some(cn) => cn,
+            }
+        };
+        let header = Header::new(header_row, Some(column_names));
```

Comment:
```suggestion
        let (column_names, header): (Vec<String>, Header)  = {
            match column_names {
                None => (Vec::from(table.columns()), Header::At(0)),
                Some(column_names) => (column_names, Header::With(column_names)),
            }
        };
```

This assumes that the column names are the first row of the table data. If they aren't, we should always use `Header::With`

Reply:
One problem with this is that usually for tables the header row is generally defined in its metadata, I've updated this logic to adhere to that, let me know what you think.

Reply:
Looks good, thanks!